internal/helpers/helpers.go

```diff
@@ -318,3 +323,43 @@ func TaskStepToStatusCondition(c string, t time.Time) metav1.Condition {
 		LastTransitionTime: metav1.NewTime(t),
 	}
 }
+
+// there is currently no reason to check the `lagoon-platform-env` secret, just the lagoon-env configmap or lagoon-env secret
+// as it contains values that are provided back to the API at the completion of a build/buildstep
+func GetLagoonEnvRoutes(ctx context.Context, opLog logr.Logger, c client.Client, namespace string) (bool, string, []string) {
```

Comment:
The whole thing looks good, I just want to raise a question about the return types here. 
I think that having typed error might be _slightly_ better semantics for the function than a return value.

Rather have the calling function check if there's an error, and if that error is one we know about - and let the calling code deal with logging?

I'm happy to approve anyways, because this makes sense - I just think typed errors might give us a little more flexibility/insight in the client code if we ever needed it.

Reply:
Ah yeah, rather than returning bool, return error and check if error?

Reply:
Yep

Reply:
I can do that :)

Reply:
Updated with change